File: side-projects/04-obfuscation-playground/examples/js/caesar.js
// Caesar cipher: encode a message then decode it back.
function shiftChar(ch, k) {
  if (ch >= "a" && ch <= "z") {
    const base = "a".charCodeAt(0);
    return String.fromCharCode(((ch.charCodeAt(0) - base + k) % 26) + base);
  }
  return ch;
}

function encode(message, k) {
  let out = "";
  for (const c of message) out += shiftChar(c, k);
  return out;
}

const plain = "the quick brown fox";
const secret = encode(plain, 3);
const back = encode(secret, 23);
console.log("plain ", plain);
console.log("secret", secret);
console.log("back  ", back);
console.log("ok", back === plain);

File: side-projects/04-obfuscation-playground/examples/python/caesar.py
# Caesar cipher: encode a message, then decode it back.
def shift_char(ch, k):
    if "a" <= ch <= "z":
        base = ord("a")
        return chr((ord(ch) - base + k) % 26 + base)
    return ch


def encode(message, k):
    return "".join(shift_char(c, k) for c in message)


def decode(message, k):
    return encode(message, 26 - k)


plain = "the quick brown fox"
secret = encode(plain, 3)
back = decode(secret, 3)
print("plain ", plain)
print("secret", secret)
print("back  ", back)
print("ok", back == plain)
